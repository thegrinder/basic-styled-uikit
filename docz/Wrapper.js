import React from 'react';
import { node } from 'prop-types';
import { ThemeProvider } from 'styled-components';
import { ModeProvider } from '../src/components/Mode';
import ButtonProvider from '../src/components/Button/ButtonProvider';
import theme from '../src/theme/theme';

const propTypes = {
  children: node.isRequired,
};

const Wrapper = ({ children }) => (
  <ModeProvider>
    <ThemeProvider theme={theme}>
      <ButtonProvider>
        <div>{children}</div>
      </ButtonProvider>
    </ThemeProvider>
  </ModeProvider>
);

Wrapper.propTypes = propTypes;

export default Wrapper;
