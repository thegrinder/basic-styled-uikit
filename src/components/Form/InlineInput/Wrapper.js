import styled from 'styled-components';

const Wrapper = styled.div`
  position: relative;
`;

export default Wrapper;
