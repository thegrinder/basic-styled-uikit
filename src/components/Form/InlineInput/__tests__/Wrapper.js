import React from 'react';
import { render, cleanup } from 'react-testing-library';
import 'jest-dom/extend-expect';

import Wrapper from '../Wrapper';

const children = <span>children</span>;
const renderComponent = () => render(
  <Wrapper>{children}</Wrapper>,
);

describe('<Wrapper />', () => {
  afterEach(cleanup);

  it('should render correctly with children', () => {
    const { container: { firstChild }, getByText } = renderComponent();
    const childrenElement = getByText('children');
    expect(firstChild).toBeDefined();
    expect(firstChild).toContainElement(childrenElement);
    expect(firstChild).toMatchSnapshot();
  });

  it('should render <div> tag', () => {
    const { container: { firstChild } } = renderComponent();
    expect(firstChild.tagName).toEqual('DIV');
  });
});
