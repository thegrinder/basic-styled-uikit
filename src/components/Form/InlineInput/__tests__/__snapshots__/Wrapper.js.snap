// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Wrapper /> should render correctly with children 1`] = `
<div
  class="Wrapper-sc-1e6m9j8-0 gUPYuR"
>
  <span>
    children
  </span>
</div>
`;
